libvast/src/arrow_table_slice.cpp

```diff
@@ -33,26 +29,6 @@
 
 namespace vast {
 
-auto values(const type& type,
-            const std::same_as<arrow::Array> auto& array) noexcept
-  -> detail::generator<data_view> {
-  const auto f = []<concrete_type Type>(
-                   const Type& type,
-                   const arrow::Array& array) -> detail::generator<data_view> {
-    for (auto&& result :
-         values(type, caf::get<type_to_arrow_array_t<Type>>(array))) {
-      if (!result)
-        co_yield {};
-      else
-        co_yield std::move(*result);
-    }
-  };
-  return caf::visit(f, type, detail::passthrough(array));
-}
```

Comment:
There was a duplicate declaration in the header, and it's also used in the header in another templated function.

To be honest I'm not exactly sure how GCC arrived at this conclusion, since after all as you said it was only _defined_ multiple times, but alas, this causes GCC not to complain about it anymore when building the Docker image.

Reply:
Yeah, weird, I thought that all templates were implicitly `inline` anyways. But not weird enough to lose sleep over :D

Reply:
I'm not exactly happy with it either and would like to understand, but it's not worth diving deeper I think :( :D